### PR TITLE
Refactor the repo, and new docs styling

### PR DESCRIPTION
**Pull request by Archmonger**
- Refactors the repo structure to match `reactpy-django`
- Update styling/formatting of the docs
 
This PR is mostly folder and CI workflow reorganization. The the docs still need an overhaul, and the source code needs refactoring to match `react-router` better.

### CODE REVIEW

**Comment by Archmonger** [comment]
@rmorshea Can you provide me merge permissions for this repo?

**Comment by Archmonger** [comment]
@rmorshea I still don't have permissions to change required workflows.

File: requirements/check-style.txt [hunk 1]
```diff
@@ -1,5 +1,2 @@
 black
```
Comment (rmorshea) [inline requirements/check-style.txt:1]:
```suggestion
black=23
```

**Comment by rmorshea** [comment]
Looks like we're missing coverage enforcement from the tests.

**Comment by Archmonger** [comment]
Will work on that now. On a related note, I've been debating whether we should drop `black` and instead rely strictly on the ruff formatter.

**Comment by Archmonger** [comment]
Will merge this one with the coverage fixes. Let me know your opinion on pulling out `black` from our repos and relying solely on `ruff`.

**Pull request merged**